File: BarberBooking.API/Dto/DtoUsers/DtoCreateUser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace BarberBooking.API.Dto.DtoUsers
{
    public class DtoCreateUser
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public DtoPhone Phone { get; set; }
        [Required]
        public string Email {get ; set;}
        [Required]
        public string PasswordHash { get; set; }
        [Required]
        public string City { get; set; }
        [Required]
        public string Devices {get; set;}
    }
}

File: barber_booking_app/lib/models/Dto/DtoCreateUser.dart
import 'package:barber_booking_app/models/base/json_serializable.dart';
import 'package:barber_booking_app/models/vo_models/dto_phone.dart';

class DtoCreateuser extends JsonSerializable {
    String? name;
    DtoPhone? phone;
    String? email;
    String? password;
    String? city;
    
    DtoCreateuser({
    this.name,
    this.phone,
    this.email,
    this.password,
    this.city
  });
  
  @override
  Map<String, dynamic> toJson() {
    return{
    'name': name,
    'phone': phone?.toJson(),
    'email': email,
    'passwordHash': password,
    'city':city
    };
  }
}